src/apic-extension/azext_apic_extension/aaz/latest/apic/api/deployment/_create.py

```diff
@@ -44,20 +44,22 @@ def _build_arguments_schema(cls, *args, **kwargs):
         # define Arg Group ""
 
         _args_schema = cls._args_schema
-        _args_schema.api_name = AAZStrArg(
-            options=["--api", "--api-name"],
-            help="The name of the API.",
+        _args_schema.api_id = AAZStrArg(
+            options=["--api-id"],
+            help="The id of the API.",
             required=True,
             fmt=AAZStrArgFormat(
+                pattern="^[a-zA-Z0-9-]{3,90}$",
                 max_length=90,
                 min_length=1,
             ),
         )
-        _args_schema.deployment_name = AAZStrArg(
-            options=["--name", "--deployment", "--deployment-name"],
```

Comment:
We need to tell users the parameter is asking for a deployment id to reduce user faults. The root cause of many bugs is user provides the name (title) of the deployment, not id.
After we change the options to id, if we add `--id`, this will cause confusions with the `--ids` parameter provided by Azure CLI, because they represent different things. I'm not sure whether `--deployment` is needed, as it doesn't tell what information it's asking. And the `az apim` commands also just have `--api-id` (without `--api` option).
But feel free to create feedback bugs to CLI so Julia and I can follow up. Resolving for now after provided above information.